### PR TITLE
 pingpong: Add support of large addresses

### PR DESCRIPTION
**Pull request by gladkovdmitry17**
The current limit of the addresses is 64 bytes that is too short for the MLX provider (UCX wrapper) - we're going to revive this provider.
This patch strives to avoid this limit by requesting the needed length of memory space that is needed for the address (`fi_getname` with the passed length = 0). After that the needed buffer is dynamically allocated and passes to the `fi_getname` again to retrieve the address.

Only UDP wasn't compliant with expected behavior. The UDP `fi_getname` has been aligned with correct behavior in this patch

### CODE REVIEW

**Comment by gladkovdmitry17** [comment]
@shefty Do you agree with my thoughts regarding `fi_getname` behavior? Should it work as expected in this PR?

File: prov/udp/src/udpx_ep.c [hunk 1]
```diff
@@ -56,11 +56,25 @@ int udpx_setname(fid_t fid, void *addr, size_t addrlen)
...
+
+	if (*addrlen < sock_addr_len)
+		ret = -FI_ETOOSMALL;
+
+	*addrlen = sock_addr_len;
+
+	if (addr)
+		memcpy(addr, &sock_addr, *addrlen);
+
+	return ret;
```
Comment (shefty) [inline prov/udp/src/udpx_ep.c]: Please separate this change into its own patch, so that the fix can be cherry-picked into stable branches.

File: prov/udp/src/udpx_ep.c [hunk 2]
```diff
@@ -56,11 +56,25 @@ int udpx_setname(fid_t fid, void *addr, size_t addrlen)
...
 	ep = container_of(fid, struct udpx_ep, util_ep.ep_fid.fid);
-	ret = getsockname(ep->sock, addr, (socklen_t *)addrlen);
-	return ret ? -errno : 0;
+	ret = getsockname(ep->sock, (struct sockaddr *)&sock_addr,
+			  (socklen_t *)&sock_addr_len);
+	if (ret)
+		return -ofi_sockerr();
+
+	if (*addrlen < sock_addr_len)
+		ret = -FI_ETOOSMALL;
```
Comment (shefty) [inline prov/udp/src/udpx_ep.c]: We can probably simplify the changes to:
```
size_t buflen = *addrlen;

ret = getsockname(ep->sock, addr, (socklen_t *) addrlen);
if (ret)
    return -errno;

return buflen < *addrlen ? -FI_ETOOSMALL : 0;
```

File: prov/udp/src/udpx_ep.c (hunk 1 above)
Comment (shefty) [reply prov/udp/src/udpx_ep.c]: Never mind -- I thought I was looking at a single commit, rather than the change set.

Comment (gladkovdmitry17) [reply prov/udp/src/udpx_ep.c]: :)

File: prov/udp/src/udpx_ep.c (hunk 2 above)
Comment (gladkovdmitry17) [reply prov/udp/src/udpx_ep.c]: Sure, thanks!

Comment (gladkovdmitry17) [reply prov/udp/src/udpx_ep.c]: I've updated the PR. But it doesn't work on Windows. Please, don't merge. I'll fix

**Comment by gladkovdmitry17** [comment]
@shefty Please, don't merge. The `upd` provider doesn't work on Windows. It crashes in `util_main.c`, because global `lock` wasn't properly initialized for Windows. Will fix this tomorrow.

**Comment by shefty** [comment]
Let me know when you're ready. Changes look okay.

**Comment by gladkovdmitry17** [comment]
> Let me know when you're ready. Changes look okay.

@shefty Now it works on Windows. I've aligned the behavior of the Win's `getsockname` with BSD `getsockname` by adding the `ofi_getsockname` abstraction.
Could you take a look, please? Thanks!

**Comment by shefty** [comment]
Looks good - thanks. Only waiting on CI prior to merging.
